seed/tests/test_ubids.py

```diff
@@ -771,3 +768,267 @@ def test_jaccard(self):
+            'property': self.child_property_state.id
+        })
+        response = self.client.post(url, params, content_type='application/json')
+        assert response.status_code == 404
+
+        # root can
+        self.login_as_root_member()
+        params = json.dumps({
+            'access_level_instance_id': self.root_level_instance.id,
+            'ubid': "D+D-0-0-0-0",
+            'property': self.child_property_state.id
+        })
+        response = self.client.post(url, params, content_type='application/json')
+        assert response.status_code == 201
+
+        # Taxlots
+        url = reverse_lazy('api:v3:ubid-list') + "?organization_id=" + str(self.org.id)
+
+        self.login_as_child_member()
+        params = json.dumps({
+            'access_level_instance_id': self.root_level_instance.id,
+            'ubid': "C+C-0-0-0-0",
+            'taxlot': self.child_taxlot_state.id
+        })
+        response = self.client.post(url, params, content_type='application/json')
+        assert response.status_code == 404
+
+        self.login_as_root_member()
+        response = self.client.post(url, params, content_type='application/json')
+        assert response.status_code == 201
+
+    def test_ubids_destroy(self):
+        url = reverse_lazy('api:v3:ubid-detail', args=[self.root_ubid.pk]) + "?organization_id=" + str(self.org.id)
+
+        self.login_as_child_member()
+        response = self.client.delete(url, content_type='application/json')
+        assert response.status_code == 404
+
+        url = reverse_lazy('api:v3:ubid-detail', args=[self.root_ubid.pk]) + "?organization_id=" + str(self.org.id)
+        self.login_as_root_member()
+        response = self.client.delete(url, content_type='application/json')
+        assert response.status_code == 204
+
+    def test_ubids_put(self):
+        url = reverse_lazy('api:v3:ubid-detail', args=[self.root_ubid.pk]) + "?organization_id=" + str(self.org.id)
+        params = json.dumps({})
+
+        self.login_as_child_member()
+        response = self.client.put(url, params, content_type='application/json')
+        assert response.status_code == 404
+
+        self.login_as_root_member()
+        response = self.client.put(url, params, content_type='application/json')
+        assert response.status_code == 200
+
+    def test_ubids_list(self):
+        url = reverse_lazy('api:v3:ubid-list') + "?organization_id=" + str(self.org.id)
+
+        # child user gets only child ubids
+        self.login_as_child_member()
+        response = self.client.get(url, content_type='application/json')
+        assert response.status_code == 200
+        assert len(response.json()['data']) == 4
+
+        # root users gets all ubids
+        self.login_as_root_member()
+        response = self.client.get(url, content_type='application/json')
+        assert response.status_code == 200
+        assert len(response.json()['data']) == 6
+
+    def test_ubids_decode_by_ids(self):
```

Comment:
this shouldn't 404, it should 200 but not return the properties/taxlot you don't have access to, same as it would ids to properties that don't exist.